### PR TITLE
pivotal/acc-content#32 - Add a fragment capable to add a AppSSO/OIDC authorization flow integration to a SpringBoot-based accelerator.

### PR DESCRIPTION
**Pull request by dmbelyaev**

### CODE REVIEW

File: spring-boot-app-sso-auth-code-flow/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Spring Boot AppSSO Authorization Code Flow Fragment
+
+A Fragment that include AppSSO OIDC Authorization Code Flow into your Spring Boot based Accelerator. This enables a Spring Boot application to support authentication delegated to an AppSSO AuthServer.
```
Comment (ericbottard) [inline spring-boot-app-sso-auth-code-flow/README.md]:
```suggestion
A Fragment that includes AppSSO OIDC Authorization Code Flow into your Spring Boot based Accelerator. This enables a Spring Boot application to support authentication delegated to an AppSSO AuthServer.
```

File: spring-boot-app-sso-auth-code-flow/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+echo "CLIENT_SECRET: $CLIENT_SECRET"
+echo "ISSUER_URI: $ISSUER_URI"
+curl -XPOST "$ISSUER_URI/oauth2/token?grant_type=client_credentials&scope=openid" -u "$CLIENT_ID:$CLIENT_SECRET"
+```
+
+> If you start a generated application locally please be sure that `local` profile is active. 
+
+## Using the fragment
+
+This fragment requires one parameter `packageName` which defines root package of an application to be generated. If your accelerator doesn't have this option you can include it by importing this fragment in your `accelerator.yaml`:
```
Comment (ericbottard) [inline spring-boot-app-sso-auth-code-flow/README.md]: This sentence can be misunderstood as "if you already have this option, you don't need to import this fragment", which is false.

File: spring-boot-app-sso-auth-code-flow/accelerator.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            oldPackageName: "'com.vmware.tap.fragments.appssoauthcodeflow'"
+            newPackageName: "#packageName"
+
+    - include: [ "pom.xml" ]
+      chain:
+        - type: ReplaceText
+          substitutions:
+            - text: "<dependencies>"
+              with: |
+                '<dependencies>
```
Comment (ericbottard) [inline spring-boot-app-sso-auth-code-flow/accelerator.yaml:28]: Why not add the 3 dependencies in one go?

File: spring-boot-app-sso-auth-code-flow/accelerator.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        <!-- AppSSO/OIDC Integration -->
+                        <dependency>
+                            <groupId>org.springframework.boot</groupId>
+                            <artifactId>spring-boot-starter-oauth2-client</artifactId>
+                        </dependency>'
+
+    - include: [ "build.gradle.kts" ]
+      chain:
+        - type: ReplaceText
```
Comment (ericbottard) [inline spring-boot-app-sso-auth-code-flow/accelerator.yaml:46]: ditto

File: spring-boot-app-sso-auth-code-flow/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# Spring Boot AppSSO Authorization Code Flow Fragment
+
+A Fragment that include AppSSO OIDC Authorization Code Flow into your Spring Boot based Accelerator. This enables a Spring Boot application to support authentication delegated to an AppSSO AuthServer.
+
+It adds Spring Security OAuth 2.0 and Thymeleaf dependencies on your `pom.xml` or `build.gradle.kts`.
+It includes `WebSecurityConfiguration` class that configures Spring Security to use OAuth 2.0.
+
+Include an additional Spring MVC Endpoint at `/protected/profile` to trigger the authorization flow and, when succesful, show the logged-in subject.
```
Comment (ericbottard) [inline spring-boot-app-sso-auth-code-flow/README.md]:
```suggestion
It includes an additional Spring MVC Endpoint at `/protected/profile` to trigger the authorization flow and, when succesful, show the logged-in subject.
```

File: spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/web/OidcUserController.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class OidcUserController {
+
+    @GetMapping("/protected/profile")
```
Comment (trisberg) [inline spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/web/OidcUserController.java:12]: The `WebSecurityConfiguration.java` file uses tabs and this one spaces for indents, should be consistent

**Review by trisberg — CHANGES_REQUESTED** [review]

File: spring-boot-app-sso-auth-code-flow/README.md (hunk 1 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/README.md]: Done

File: spring-boot-app-sso-auth-code-flow/accelerator.yaml (hunk 3 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/accelerator.yaml:28]: Done

File: spring-boot-app-sso-auth-code-flow/accelerator.yaml (hunk 4 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/accelerator.yaml:46]: Done

File: spring-boot-app-sso-auth-code-flow/README.md (hunk 5 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/README.md]: Done

File: spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/web/OidcUserController.java (hunk 6 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/web/OidcUserController.java:12]: Done

File: spring-boot-app-sso-auth-code-flow/README.md (hunk 2 above)
Comment (dmbelyaev) [reply spring-boot-app-sso-auth-code-flow/README.md]: Done

**Review by trisberg — APPROVED** [review]

**Comment by trisberg** [comment]
We need to have a consistent policy across accelerators and fragments for Java source indentation - tabs or spaces - we have a mix right now. We can address this in a separate issue once we reach consensus.

**Pull request merged**